guiUtils/guiUtils.js

```diff
@@ -0,0 +1,26 @@
+function filterByTestName(search, testName) {
+	if (testName) {
+		if (search) {
+			if (!search.match(/.*testNameFilter=.*/)) {
+				return search+='&testNameFilter='+testName;
+			} else {
+				return search.replace(/testNameFilter=[^&]*/, 'testNameFilter='+testName);
+			}
+		} else {
+			return '?testNameFilter='+testName;
```

Comment:
This is super complicated because you're trying to cover all cases. Just remove the currently set filter, and set the new one:

```js
let result = search.replace(/testNameFilter(=[^&]+)?/, '');
return `result${result.indexOf('?') == -1 ? '?' : '&')}testNameFilter=${testName}`;
```